main/realmain.cc

```diff
@@ -540,6 +540,7 @@ int realmain(int argc, char *argv[]) {
         }
 
         inputFiles = pipeline::reserveFiles(*gs, opts.inputFileNames);
+        opts.inputFileNames = {};
```

Comment:
```suggestion
        // This vector + string storage can be 10s of MB on large codebases,
        // and we don't use it past this point.  Clear it to save memory.
        opts.inputFileNames.clear();
```

Maybe a comment too, on why we're doing this?